### PR TITLE
#166855378 ; shuts down the viewer when data is no longer streaming

### PR DESCRIPTION
**Pull request by lawhead**

### CODE REVIEW

**Comment by lawhead** [comment]
https://www.pivotaltracker.com/n/projects/2119090

**Comment by orbitalhybridization** [comment]
I was able to replicate the bug with your instructions, thanks! After cloning the branch and trying to replicate the bug again with the new changes, the viewer window flashes for a second, closes, and the process terminates with the output:

Waiting for LSL EEG data stream...
Sample rate: 256.0
Fast forwarding:
Chomped 0 records.

Is this expected? And has anyone else gotten this behaviour? I might be doing something wrong, but am not sure...

- Ian

[Edit: formatting]

**Comment by lawhead** [comment]
Ian, thanks for testing. Yes, that is the expected output.

**Review by tab-cmd — APPROVED** [review]

File: bcipy/gui/viewer/data_source/lsl_data_source.py [hunk 1]
```diff
@@ -60,7 +59,7 @@ def next_n(self, n: int, fast_forward=False):
             print(f'Chomped {chomped_count} records.')
             samples = samples[len(tmp):] + tmp            
 
-        if len(samples) < n:
+        if len(samples) < n and not fast_forward:
```
Comment (tab-cmd) [inline bcipy/gui/viewer/data_source/lsl_data_source.py:71]: A comment here or an addition to the docstring for the method would be helpful.

File: bcipy/gui/viewer/data_viewer.py [hunk 2]
```diff
@@ -318,6 +318,9 @@ def update_buffer(self, fast_forward=False):
                 self.buffer.append(row)
         except StopIteration:
             self.stop()
+            self.Close()
```
Comment (tab-cmd) [inline bcipy/gui/viewer/data_viewer.py:322]: Is this closing Wx.Frame? We could have a `close` or `exit` method that wraps this which might be more clear

**Review by tab-cmd — APPROVED** [review]

**Pull request merged**